examples/squiral/main.go

```diff
@@ -0,0 +1,371 @@
+		dir := (s.dir + next) % 4
+		off := dirs[dir]
+		// Peek all targets by priority.
+		target := tup{
+			x: x + off.x,
+			y: y + off.y,
+		}
+		if auto.colorMap[target.x][target.y] == free {
+			// If the target is free we need to also check the
+			// surrounding cells.
+
+			// a. Test if next cell in direction dir does not have
+			// the same color as this squiral.
+			ntarg := tup{
+				x: target.x + off.x,
+				y: target.y + off.y,
+			}
+			if auto.colorMap[ntarg.x][ntarg.y] == s.col {
+				// If this has the same color, we cannot go into this direction,
+				// to avoid ugly blocks of equal color.
+				continue // try next direction
+			}
+
+			// b. Test all outer fields for the color of the
+			// squiral itself.
+			horivert := dir % 2
+			xtarg := tup{}
+			set := true
+			for _, out := range outers[horivert] {
+				xtarg.x = target.x + out.x
+				xtarg.y = target.y + out.y
+
+				// If one of the outer targets equals the squiral's
+				// color, again continue with next direction.
+				if auto.colorMap[xtarg.x][xtarg.y] == s.col {
+					// If this is not free we cannot go into this direction.
+					set = false
+					break // try next direction
+				}
+
+				xtarg.x = ntarg.x + out.x
+				xtarg.y = ntarg.y + out.y
+
+				// If one of the outer targets equals the squiral's
+				// color, again continue with next direction.
+				if auto.colorMap[xtarg.x][xtarg.y] == s.col {
+					// If this is not free we cannot go into this direction.
+					set = false
+					break // try next direction
+				}
+			}
+
+			if set {
+				s.pos = target
+				s.dir = dir
+				// 2. set the color of this squiral to its
+				// current position.
+				setpix(s.pos, s.col)
+				return
+			}
+		}
+	}
+
+	s.dead = true
+}
+
+type automaton struct {
+	squirals [amount]squiral
+	colorMap [width][height]color.Color
+}
+
+func (au *automaton) init() {
+	// Init the test grid with color (0,0,0,0) and the borders of
+	// it with color(0,0,0,254) as a blocker color, so the squirals
+	// cannot escape the scene.
+	for x := 0; x < width; x++ {
+		for y := 0; y < height; y++ {
+			if x == 0 || x == width-1 || y == 0 || y == height-1 {
+				auto.colorMap[x][y] = blocker
+			} else {
+				auto.colorMap[x][y] = free
+			}
+		}
+	}
+
+	for i := 0; i < amount; i++ {
+		auto.squirals[i].spawn()
+	}
+}
+
+func (au *automaton) step() {
+	for i := 0; i < amount; i++ {
+		for s := 0; s < au.squirals[i].speed; s++ {
+			au.squirals[i].step(i)
+			if au.squirals[i].dead {
+				au.squirals[i].spawn()
+			}
+		}
+	}
+}
+
+func setpix(xy tup, col color.Color) {
+	canvas.Set(xy.x, xy.y, col)
+	auto.colorMap[xy.x][xy.y] = col
+}
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+	c, err := ebiten.NewImage(width, height, ebiten.FilterNearest)
+	if err != nil {
+		panic(err)
+	}
+	canvas = c
+	canvas.Fill(colornames.Black)
+
+	auto = &automaton{}
+	auto.init()
+}
+
+func update(screen *ebiten.Image) error {
+	reset := false
+
+	if ebiten.IsKeyPressed(ebiten.KeyEscape) {
```

Comment:
As the window can be closed by `Alt+F4` on Windows and `Command + Q` on Mac, I don't think we need to add this.

Reply:
Usually I really prefer to quit demos with `esc` over `alt+f4` but it's not really important so I will remove it.

Reply:
I understand your idea, but for example, there is no 'termination' on browsers or mobiles. Then, not having a way to terminate the app is more portable.

Reply:
I see. That's a good argument! :)